MSSQL/OMOPLoader.sql

```diff
@@ -1613,43 +1613,93 @@ IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'[dbo].[build
+declare @temp_tableidx varchar(100)
 
+--Cursors
 declare getconfigsql cursor local for
 select c_tablename, c_path, c_target_column from i2o_config_modifier where c_domain=@domain
 
 begin
+	--Loop Cursor
     open getconfigsql;
     fetch next from getconfigsql into @tablename,@path,@target_column;
     while @@fetch_status=0
     begin
-        set @sqltext = 'select patient_num, encounter_num, m.provider_id, concept_cd, start_date, o.i_stdcode  ' +@target_column+' ' +
-            'into temp_mod_'+@target_column+' from i2b2fact M ' +
-            --inner join visit_occurrence enc on enc.person_id = m.patient_num and enc.visit_occurrence_id = m.encounter_Num
-            ' inner join '+@tablename+' o on m.modifier_cd =o.c_basecode ' +
-            ' where c_fullname LIKE '''+@path+ '%'''
-        exec sp_executesql @sqltext
-        raiserror('Modifiers: %s',0,1,@sqltext) with NOWAIT;
+
+		--Set table temp name and idx name
+		set @temp_tablename = 'temp_mod_' + @target_column;
+		set @temp_tableidx = @temp_tablename + '_cidx';
+
+		-- If force_update then drop existing temp table if it exists
+		IF @force_update = 1 
+			BEGIN
+					IF OBJECT_ID(@temp_tablename) IS NOT NULL
+						BEGIN TRY
+							set @sqltext = 'DROP TABLE ' + @temp_tablename;
+							exec sp_executesql @sqltext;
+						END TRY
+						BEGIN CATCH
+							RAISERROR('ERROR: Failed to execute %s', 0, 1, @sqltext) with NOWAIT;
+						END CATCH;
+			END;
+
+		-- If the temp table does not already exist create it and index it
+		IF OBJECT_ID(@temp_tablename) IS NULL
+			BEGIN TRY
+				set @sqltext = 'select patient_num, encounter_num, m.provider_id, concept_cd, start_date, instance_num, o.i_stdcode  ' +@target_column+' ' +
+					'into temp_mod_'+@target_column+' from i2b2fact M ' +
+					' inner join '+@tablename+' o on m.modifier_cd =o.c_basecode ' +
+					' where c_fullname LIKE '''+@path+ '%'''
+				exec sp_executesql @sqltext
+				set @sqltext = 'create clustered index ' + @temp_tableidx + ' on ' + @temp_tablename + '(patient_num, encounter_num, provider_id, concept_cd, start_date, instance_num)'
+				exec sp_executesql @sqltext
+				PRINT 'CREATED TEMPORARY TABLE ' + @TEMP_TABLENAME;
+			END TRY
+			BEGIN CATCH
+				raiserror('Error: Failed to execute: %s',0, 1, @sqltext) with NOWAIT;
+			END CATCH;
+
         fetch next from getconfigsql into @tablename,@path,@target_column;
     end
+	--Close Loop
     close getconfigsql;
     deallocate getconfigsql;
 end
-GO
 
 --exec build_modifiers lab
 --select patient_num, encounter_num, m.provider_id, concept_cd, start_date, o.i_stdcode  priority into #priority from i2b2fact M  inner join pcornet_lab o on m.modifier_cd =o.c_basecode  where c_fullname LIKE '\PCORI_MOD\PRIORITY\%'
 
 IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'OMOPmeasurementLab') AND type in (N'P', N'PC')) DROP PROCEDURE OMOPmeasurementLab;
 GO
-create procedure OMOPmeasurementLab as
+create procedure [dbo].[OMOPmeasurementLab] as

```

Comment:
Same comment, I don't think we want the dbo unless it needs to be there.

Reply:
Removed schema reference